lib/services/elasticsearch.js

```diff
@@ -153,519 +166,479 @@ class ElasticSearch extends Service {
+          : Bluebird.resolve({ body: false });
 
-              // The document exits and is active, we reject to prevent the creation
-              return this.reject('document_already_exists', esRequest.id);
-            })
-            // Pitfall of all previous rejections
-            .catch(err => {
-              if (err.displayName === 'NotFound') {
-                // The document doesn't exist, we create it
-                return this.client.create(esRequest)
-                  .then(result => this.refreshIndexIfNeeded(
-                    esRequest,
-                    _.extend(result, {_source: request.input.body}))
-                  )
-                  .catch(error => Bluebird.reject(
-                    this.esWrapper.formatESError(error))
-                  );
-              }
-
-              // A "real" error occurred, we reject it
-              return Bluebird.reject(err);
-            });
+        return promise;
+      })
+      .then(({ body: exists }) => {
+        if (exists) {
+          return this.reject('document_already_exists', esRequest.id);
         }
 
-        return this.client.index(esRequest)
-          .then(result => this.refreshIndexIfNeeded(
-            esRequest,
-            _.extend(result, {_source: request.input.body}))
-          )
-          .catch(error => Bluebird.reject(
-            this.esWrapper.formatESError(error))
-          );
-      });
-  }
+        return this.client.index(esRequest);
+      })
+      .then(({ body }) => {
+        const apiResponse = {
+          _id: body._id,
+          _version: body._version,
+          _source: esRequest.body
+        };
+
+        return apiResponse;
+      })
+      .catch(error => this.esWrapper.reject(error));
+}
 
   /**
    * Create a new document to ElasticSearch, or replace it if it already exist
    *
-   * @param {Request} request
-   * @param {boolean} injectKuzzleMeta
-   * @returns {Promise} resolve an object that contains _id
+   * @param {string} index - Index name
+   * @param {string} collection - Collection name
+   * @param {string} id - Document id
+   * @param {object} content - Document content
+   * @param {object} options - refresh (null), userId (null), injectKuzzleMeta (false)
+   *
+   * @returns {Promise} { _id, _version, _source, created }
    */
-  createOrReplace(request, injectKuzzleMeta = true) {
-    const
-      esRequest = initESRequest(request, ['refresh']),
-      userId = this._getUserId(request),
-      kuzzleMeta = {
-        author: userId,
-        createdAt: Date.now(),
-        updatedAt: Date.now(),
-        updater: userId,
-        active: true,
-        deletedAt: null
-      };
-
-    esRequest.id = request.input.resource._id;
-    esRequest.body = request.input.body;
+  createOrReplace (
+    index,
+    collection,
+    id,
+    content,
+    { refresh=null, userId=null, injectKuzzleMeta=false })
+  {
+    const esRequest = {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Group all shorthand properties at the end of this object declaration [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3499)